src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +153,97 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        private void DetectScriptLocation()
+        {
+            if (_scriptDetectionFailed || AutoConfigUrl != null)
+            {
+                return;
+            }
+
+            WinHttpTraceHelper.Trace("WinInetProxyHelper.DetectScriptLocation: Attempting discovery PROXY_AUTO_DETECT_TYPE_DHCP.");
+            AutoConfigUrl = GetAutoProxyConfigUrl(Interop.WinHttp.AutoDetectType.Dhcp);
+
+            if (AutoConfigUrl == null)
+            {
+                WinHttpTraceHelper.Trace("WinInetProxyHelper.DetectScriptLocation: Attempting discovery AUTO_DETECT_TYPE_DNS_A.");
+                AutoConfigUrl = GetAutoProxyConfigUrl(Interop.WinHttp.AutoDetectType.DnsA);
```

Comment:
You don't need to call this twice.  The AutoDetectType enum values are flags.  So, just OR them together and use a single call to WinHttpDetectAutoProxyConfigUrl.

Reply:
I copied the pattern from Framework, and I think this approach is slightly better than OR them together: assume `WinHttpDetectAutoProxyConfigUrl` succeed, by separating the operations, we will know exactly which protocol is used to locate the PAC file, and log the information.

Reply:
It is most likely going to be slower to make separate calls.  And the only benefit is being able to log more details.  I don't think that is a sufficient reason to make the product code slower all the time.

Reply:
Will change.